File: code/jvm/repository/src/main/java/pt/isel/repositories/users/UserRepository.java
package pt.isel.repositories.users;

import pt.isel.domain.security.PasswordValidationInfo;
import pt.isel.domain.users.User;
import pt.isel.repositories.Repository;

public interface UserRepository extends Repository<User> {
    User create(String username, PasswordValidationInfo passwordValidationInfo);

    User findByUsername(String username);

    boolean hasUsers();
}

File: code/jvm/repository/src/main/kotlin/pt/isel/repositories/users/UserRepository.kt
package pt.isel.repositories.users

import pt.isel.domain.security.PasswordValidationInfo
import pt.isel.domain.users.User
import pt.isel.repositories.Repository

/** Repository interface for managing users, extends the generic Repository */
interface UserRepository : Repository<User> {
    fun create(
        username: String,
        passwordValidationInfo: PasswordValidationInfo,
    ): User

    fun findByUsername(username: String): User?

    fun hasUsers(): Boolean
}
